templates/react-with-storybook/.storybook/main.js

```diff
@@ -1,4 +1,7 @@
 module.exports = {
   stories: ['../stories/**/*.stories.@(ts|tsx|js|jsx)'],
   addons: ['@storybook/addon-links', '@storybook/addon-essentials'],
+  typescript: {
+    check: true, // set to false if you don't care about your stories being type-checked.
```

Comment:
```suggestion
    check: true, // type-check stories during Storybook build
```

"if you don't care" is really strong wording that I think we should generally avoid, but also it's not particularly helpful -- why would one not care? (a perf need is the main one). 

But rather than enumerating all reasons to not set it, let's just explain what the option does for now